pkg/core/xds/matched_rules.go

```diff
@@ -0,0 +1,97 @@
+package xds
+
+import (
+	"fmt"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
+)
+
+const ClientRuleAttachmentType = "clientSubset"
+const DestinationRuleAttachmentType = "destinationSubset"
+const SingleItemRuleAttachmentType = "singleItem"
+
+type RuleAttachment struct {
+	Type       string
+	Name       string
+	Service    string
+	PolicyType core_model.ResourceType
+	Rule       Rule
+}
+
+func BuildRulesAttachments(
```

Comment:
Interesting to put this in `pkg/core/xds` is it going to be used in other places than the api-server? Seems like you are playing with different types etc for code that's not actually going to be reused else where no?

Reply:
You can bring the same argument about `matched_policy.go`. Inspect elements and types are used only by the api-server.

Should all of this go to `pkg/core/xds/inspect`?

Reply:
Sounds like an improvement to me

Reply:
moved my stuff. I'll move other things in the next PR to not mess with this one